iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -554,8 +555,12 @@ private async Task HandleIncomingMessagesAsync()
             if (Logging.IsEnabled)
                 Logging.Enter(this, "Process C2D message via callback", nameof(HandleIncomingMessagesAsync));
 
-            Message message = ProcessMessage(true);
-            await (_deviceMessageReceivedListener?.Invoke(message) ?? TaskHelpers.CompletedTask).ConfigureAwait(false);
+            Message message = ProcessMessage();
+
+            // We are intentionally not awaiting _deviceMessageReceivedListener callback.
+            // This is a user-supplied callback that isn't required to be awaited by us. We can simply invoke it and continue.
+            _ = _deviceMessageReceivedListener?.Invoke(message);
```

Comment:
can discard assignment and rather just _deviceMessageReceivedListener?.Invoke(message);

Correct me if I am missing something.

Reply:
you could, yes, but this is to highlight that the function "_deviceMessageReceivedListener" returns a task but we intentionally don't await it